utils/python/CIME/hist_utils.py

```diff
@@ -0,0 +1,175 @@
+"""
+Functions for actions pertaining to history files.
+"""
+
+from CIME.XML.standard_module_setup import *
+
+import logging, glob, os, shutil, re
+
+def _iter_model_file_substrs(case):
+
+    models = case.get_compset_components()
+    models.append('cpl')
+    for model in models:
+        yield model
+
+def _get_all_hist_files(testcase, model, suffix="", from_dir=os.getcwd()):
```

Comment:
Here and elsewhere: Please change the way the directory defaults are done so that the default is None, and then, within the function, if the argument is None it is set to os.getcwd(). (In order to address the same problem I noted in my PR a few days ago: os.getcwd() is evaluated when the module is loaded rather than when the function is called.)

Or, better: avoid having a default for these directory arguments, and require them to be passed explicitly. I think that requiring these arguments to be explicit will reduce a possible source of errors moving forward - particularly since I suspect that the current working directory is often _not_ what you want here.